src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAOSpec.scala

```diff
@@ -63,4 +65,102 @@ class HttpGoogleServicesDAOSpec extends FlatSpec with Matchers {
     objectMetadata.name should equal("test-composite-object")
     objectMetadata.md5Hash should equal(None)
   }
+
+  {
+    val headers = List[AnyRef]("header 1", "header 2", "header 3").asJava
+
+    val row1 = List[AnyRef]("proj 1", "free", "trial").asJava
+    val row2 = List[AnyRef]("proj 2", "firecloud", "credits").asJava
+    val row3 = List[AnyRef]("proj 3", "test", "data").asJava
+
+    val cells1 = List(headers, row1, row2, row3)
+
+    it should "have no changes if the data didn't change" in {
+      check(
+        newContent = cells1,
+        existingContent = cells1,
+        expectedOutput = cells1
+      )
+    }
+
+    it should "not explode if there is no data (only headers)" in {
+      check(
+        newContent = List(headers),
+        existingContent = List(headers),
+        expectedOutput = List(headers)
+      )
+    }
+
+    it should "not explode or delete data if the update has no data" in {
+      check(
+        newContent = List(headers),
+        existingContent = cells1,
+        expectedOutput = cells1
+      )
+    }
+
+    it should "add a new row even if existing is empty" in {
+      check(
+        newContent = cells1,
+        existingContent = List(headers),
+        expectedOutput = cells1
+      )
+    }
+
+    it should "add a new row without modifying existing ones" in {
+      check(
+        newContent = List(headers, row1, row2, row3),
+        existingContent = List(headers, row1, row2),
+        expectedOutput = List(headers, row1, row2, row3)
+      )
+    }
+
+    it should "output the union of the rows (overlapping sets)" in {
+      check(
+        newContent = List(headers, row2, row3),
+        existingContent = List(headers, row1, row2),
+        expectedOutput = List(headers, row1, row2, row3)
+      )
+    }
+
+    it should "output the union of the rows (disjoint sets)" in {
+      check(
+        newContent = List(headers, row3),
+        existingContent = List(headers, row1, row2),
+        expectedOutput = List(headers, row1, row2, row3)
+      )
```

Comment:
I think these last few are order-dependent, aren't they? Might be worth mucking with the order a bit ... 

```
 check(
        newContent = List(headers, row1),
        existingContent = List(headers, row2, row3),
        expectedOutput = List(headers, row2, row3, row1)
      )
```